src/main/java/com/angelbroking/smartapi/SmartConnect.java

```diff
@@ -1,727 +1,781 @@
+
+
+    public SmartConnect(String apiKey) {
+        this.apiKey = apiKey;
+    }
+
+    public SmartConnect(String apiKey, String accessToken, String refreshToken) {
+        this.apiKey = apiKey;
+        this.accessToken = accessToken;
+        this.refreshToken = refreshToken;
+    }
+
+
+    /**
+     * Returns apiKey of the App.
+     *
+     * @return String apiKey is returned.
+     * @throws CustomException if _apiKey is not found.
+     */
+    public String getApiKey() throws CustomException {
+        if (apiKey != null) return apiKey;
+        else throw new CustomException("The API key is missing.");
+    }
+
+
+    /**
+     * Returns accessToken.
+     *
+     * @return String access_token is returned.
+     * @throws CustomException if accessToken is null.
+     */
+    public String getAccessToken() throws CustomException {
+        if (accessToken != null) return accessToken;
+        else throw new CustomException("The Access Token key is missing.");
+    }
+
+
+    /**
+     * Returns userId.
+     *
+     * @return String userId is returned.
+     * @throws CustomException if userId is null.
+     */
+    public String getUserId() {
+        return Optional.ofNullable(userId).orElseThrow(() -> new CustomException("The user ID is missing."));
+
+    }
+
+
+    /**
+     * Returns publicToken.
+     *
+     * @return String public token is returned.
+     * @throws CustomException if publicToken is null.
+     */
+    public String getPublicToken() throws CustomException {
+        if (refreshToken != null) {
+            return refreshToken;
+        } else {
+            throw new CustomException("The Public Token key is missing.");
+        }
+    }
+
+
+    /**
+     * Retrieves login url
+     *
+     * @return String loginUrl is returned.
+     */
+    public String getLoginURL() throws CustomException {
+        String baseUrl = routes.getLoginUrl();
+        if (baseUrl != null) {
+            return baseUrl;
+        } else {
+            throw new CustomException("The Login URL key is missing.");
+        }
+    }
+
+
+    /**
+     * Generates a session for the given client with the provided credentials and TOTP.
+     *
+     * @param clientCode the client code for which the session needs to be generated
+     * @param password   the password of the client account
+     * @param totp       the TOTP generated by the client's TOTP device
+     * @return a User object representing the client's session, or null if an error occurs
+     * @throws SmartAPIException if an error occurs while making the API request
+     * @throws IOException       if an I/O error occurs while making the API request
+     * @throws JSONException     if there is an error while parsing the JSON response
+     */
+    public User generateSession(String clientCode, String password, String totp) {
+        User user;
+        JSONObject loginResultObject;
+        try {
+            smartAPIRequestHandler = new SmartAPIRequestHandler(proxy);

```

Comment:
Why do we need to create this object? Why can't we inject it?

Reply:
instead of injecting have created a private global variable for SmartAPIRequestHandler class.
because the object is been used in multiple methods